src/services/identity/ReposService.ts

```diff
@@ -108,37 +115,25 @@ export default class ReposService {
     this.setUrlsInLocalConfig(dir, repoName, stagingUrl, productionUrl)
 
     // 2. Commit changes in local repo
-    await git.add({ fs, dir, filepath: "." })
-    await git.commit({
-      fs,
-      dir,
-      message: "Set URLs",
-      author: {
-        name: ISOMER_GITHUB_ORGANIZATION_NAME,
-        email: "isomeradmin@users.noreply.github.com",
-      },
-    })
+    await this.simpleGit
+      .cwd(dir)
+      .checkout("staging") // ensure on staging branch
+      .add(".")
+      .addConfig("user.name", ISOMER_GITHUB_ORGANIZATION_NAME)
+      .addConfig("user.email", ISOMER_GITHUB_EMAIL)
+      .commit("Set URLs")
 
     // 3. Push changes to staging branch
-    const remote = "origin"
-    await git.push({
-      fs,
-      http,
-      dir,
-      remote,
-      remoteRef: "staging",
-      onAuth: () => ({ username: "user", password: SYSTEM_GITHUB_TOKEN }),
-    })
+    await this.simpleGit.push("origin", "staging")
 
-    // 4. Push changes to master branch
-    await git.push({
-      fs,
-      http,
-      dir,
-      remote,
-      remoteRef: "master",
-      onAuth: () => ({ username: "user", password: SYSTEM_GITHUB_TOKEN }),
-    })
+    // 4. Merge these changes into master branch
+    await this.simpleGit.checkout("master").merge(["staging"])
+
+    // 5. Push changes to master branch
+    await this.simpleGit.push("origin", "master")
```

Comment:
Simple Git will need to always do a `.cwd(dir)` every time before performing the action.

Same for the rest of the operations in this file too.

Reply:
Oh really? Somehow, in my tests after the first time it worked fine. Let me add it in to be cautious

Reply:
Yeah unfortunately we need to add it in because of concurrency issues. There can be some other place using the same simpleGit instance, which causes the working directory to change as a result and throw a series of errors.

Reply:
sure added